### ISSUE TITLE
FPM deb build failing

### ISSUE THREAD

**Issue by drewsonne** [goreleaser/goreleaser#333]
I'm having an error with fpm building deb packages on travis. I've tried locking my fpm version to a known working one, but that doesn't seem to be working.

The error I'm getting is 
`
error={:timestamp=>"2017-08-14T06:53:54.886278+0000", :message=>"Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag", :level=>:warn}
`
https://travis-ci.org/drewsonne/go-gocd/jobs/264257111#L908

The FPM section of my config is
```yaml
fpm:
  homepage: https://github.com/drewsonne/go-gocd
  maintainer: Drew J. Sonne <drew.sonne@gmail.com>
  license: LGPLv3
  formats:
    - deb
    - rpm
  dependencies:
    - git
```

https://github.com/drewsonne/go-gocd/blob/master/.goreleaser.yml#L49

Is this an issue with configuration, or fpm/goreleaser interaction?

I'm not sure it's a bug, but I'm at a dead end as to how to fix it.

**Comment by jspc** [comment]
I'll have a look this morning

**Comment by caarlos0** [comment]
I've seen this happening in goreleaser build sometimes as well... 
AFAIK your config is right... maybe it's a bug in fpm?

**Comment by drewsonne** [comment]
I've tried pinning my install to a few different versions of fpm, but it doesn't seem to make a difference. I will be more systematic and try a few different versions of fpm.

**Comment by rliebz** [comment]
I had a similar issue trying to run goreleaser locally on my macbook. @drewsonne—the line you linked to is actually just a warning (which I still get when I run fpm manually), but the actual error is thrown here: https://travis-ci.org/drewsonne/go-gocd/jobs/264257111#L910. It looks like the warning is only printed when there is an error as well.

Locally, the fix for me was using the GNU version of tar (1.29) as opposed to the BSD version I had installed by default (2.8.3). I'm not sure whether that will help in your case, though.

**Comment by mattstratton** [comment]
I'm getting this erratically when running in travis, myself. 

https://travis-ci.org/devopsdays/devopsdays-cli/builds/284277489

**Comment by jorinvo** [comment]
We also still get this on Travis for the goreleaser repo itself. For now, all we can to is restart these builds.
In the last 7 days we got this error 3 times out of 39.
https://travis-ci.org/goreleaser/goreleaser/builds/282349303
https://travis-ci.org/goreleaser/goreleaser/builds/282331489
https://travis-ci.org/goreleaser/goreleaser/builds/282192408

**Comment by drewsonne** [comment]
@rliebz how did you specify the GNU version of tar? or are you controlling the build environment so `tar` links to `gtar`?

**Comment by rliebz** [comment]
@drewsonne This wouldn't fix it on Travis, but if you're doing things on macOS:

```bash
brew install gnu-tar
export PATH="/usr/local/opt/gnu-tar/libexec/gnubin:$PATH"
```

And just make sure you set your path in your `.bash_profile` or any time you run `fpm`.

As for Travis, it looks like from your logs that it's the `tar` step that failed, but it's not necessarily true that the reason the command failed for you is the same reason it failed for me—all the logs really indicate is that `tar` is the step that failed.

**Comment by rliebz** [comment]
After trying to automate my releases, I was running into the same error on CircleCI reproducibly. I believe in my case it was permission issues based on how I was able to fix it.

- Same error as above is here: https://circleci.com/gh/rliebz/tusk/75#config/containers/0
  - While CircleCI doesn't let me link lines, the command I ran is on line `63`
- Successful version is here: https://circleci.com/gh/rliebz/tusk/82#config/containers/0
  - Lines `64-66` are the fix I used, although someone more fluent in `bash` may know of a cleaner syntax to get similar results
- The `.goreleaser.yml` I used is the same between both of the above builds: https://github.com/rliebz/tusk/blob/v0.1.4/.goreleaser.yml

Based on the discussion above, it seems this error message can be caused by any failure while running `tar`. I've managed to trigger it for at least two different errors in two different environments. 

A solution for the general case might be to see if we can't get the `stdout`/`stderr` from the `tar` command on failures to help with debugging, since there may or may not be anything `goreleaser` can do depending on the cause of failure.

**Comment by caarlos0** [comment]
```
{:timestamp=>"2017-10-19T22:54:48.593577+0000", :message=>"tar: ./usr/local/bin/goreleaser: file changed as we read it", :level=>:info, :file=>"cabin/mixins/pipe.rb", :line=>"47", :method=>"block in pipe"}
```

https://travis-ci.org/goreleaser/goreleaser/builds/290224710#L1065

maybe is something related to the concurrency? maybe fpm doesn't handle it very well...

**Comment by caarlos0** [comment]
check #407 

I ran:

```console
while true; do
  go test -v ./pipeline/fpm/... || break
done
```

before and after my changes.

Before it broke after 5min or so.

After, I let it run for 15min and it didn't failed. I'm assuming the concurrency is the issue here :/

**Comment by tombh** [comment]
I'm using `go get -u github.com/goreleaser/goreleaser`, which I assume pulls master? But I just got this same error as described in this thread: https://travis-ci.org/tombh/texttop/builds/336471125

**Comment by caarlos0** [comment]
hey @tombh 

the error is actually different as it seems, but is also related to tar.

```
{:timestamp=>"2018-02-02T09:25:14.848077+0000", :message=>"Process failed: tar failed (exit code 1). Full command was:[\"tar\", \"-C\", \"/tmp/fpm285281649/package-dir-staging-1ae5c43da84f54176ed1a70228f380167bf5de48f38802440c15b4ece83e\", \"-z\", \"--numeric-owner\", \"--owner\", \"0\", \"--numeric-owner\", \"--group\", \"0\", \"-cf\", \"/tmp/fpm285281649/package-deb-build-e0692d684d08e0e54e0c0d79045dfbb44c9684bb5ea75f786a6bffff3fa8/data.tar.gz\", \".\"]", :level=>:error}
```

I'm working on stop using fpm for those reasons. I already have a native impl of deb but I still need to implement RPM before I can replace fpm entirely.

For now, I think you can check if [fpm issues on tar](https://github.com/jordansissel/fpm/search?q=tar&type=Issues&utf8=%E2%9C%93), there are lots of them :/

tl;dr: this is probably not an issue on goreleaser itself but rather on fpm (which goreleaser uses).

**Comment by tombh** [comment]
Oh sorry I didn't look closely enough. Woah, fpm has a lot of open tar issues! Ok, I'll wait, thank you :)
